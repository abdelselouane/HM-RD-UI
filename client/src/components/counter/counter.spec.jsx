/* global describe, beforeAll, jest, it, expect */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Counter from './index';

describe('<Counter />', () => {
  let wrapper;
  let props;
  let spy;

  spy = null;

  beforeAll(() => {
    props = {
      name: 'mockCounter',
      handleChangeValue: jest.fn()
    };

    wrapper = mount(<Counter {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleChangeValue on increment button click', () => {
    spy = jest.spyOn(props, 'handleChangeValue');
    wrapper.setProps({ value: 1 });
    wrapper.find('button.counter-increment').simulate('click');
    expect(spy).toHaveBeenCalledWith(props.name, 2);
    spy.mockReset();
    spy.mockRestore();
  });

  it('should call handleChangeValue on decrement button click', () => {
    spy = jest.spyOn(props, 'handleChangeValue');
    wrapper.setProps({ value: 2 });
    wrapper.find('button.counter-decrement').simulate('click');
    expect(spy).toHaveBeenCalledWith(props.name, 1);
    spy.mockReset();
    spy.mockRestore();
  });

  it('should not change value on input changes', () => {
    wrapper.setProps({ value: 1 });
    wrapper
      .find('.counter-value')
      .simulate('change', { target: { value: 100 } });
    expect(wrapper.prop('value')).toBe(1);
  });

  it('should have quantity with floating zero', () => {
    wrapper.setProps({ value: 1 });
    expect(wrapper.find('#mockcounter-qty').prop('value')).toBe('01');
  });

  it('should not have floating zero when the quantity is greater than 9', () => {
    wrapper.setProps({ value: 10 });
    expect(wrapper.find('#mockcounter-qty').prop('value')).toBe('10');
  });
});
