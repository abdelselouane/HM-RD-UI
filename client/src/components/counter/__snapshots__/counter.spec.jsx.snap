// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render properly 1`] = `
.c0 {
  width: 100px;
}

.c0 .counter-decrement,
.c0 .counter-increment {
  width: 30px;
}

<Styled(Counter)
  handleChangeValue={[MockFunction]}
  name="mockCounter"
>
  <Counter
    className="c0"
    handleChangeValue={[MockFunction]}
    max={10}
    min={1}
    name="mockCounter"
    value={1}
  >
    <div
      className="counter-wrapper"
    >
      <button
        className="counter-decrement"
        disabled={true}
        id="mockcounter-dec"
        onClick={[Function]}
      />
      <input
        className="counter-value"
        id="mockcounter-qty"
        onChange={[Function]}
        type="number"
        value="01"
      />
      <button
        className="counter-increment"
        disabled={false}
        id="mockcounter-inc"
        onClick={[Function]}
      />
    </div>
  </Counter>
</Styled(Counter)>
`;
